app/controllers/sellers_controller.rb

```diff
@@ -0,0 +1,33 @@
+class SellersController < ApplicationController
+  def index
+    @sellers = Seller.all
+
+    @num_products_for_sellers = hash_of_num_products_for_each_seller(@sellers)
+  end
+
+  def show
+    @seller = Seller.find(params[:id])
+  end
+
+  private
+
+  # these two methods below help calculate a seller's
+  # total number of products in stock, we can use methods
+  # like these if we want to display other things about
+  # the seller on the sellers#index page
+  def calculate_num_products(seller)
```

Comment:
I think I would refactor both of these methods into model methods. They all relate to information in the database, and putting them in the model will likely make them more generic, opening them up to reuse later.

For example, I believe you could replace the iteration startin on line 20 with a model scope. Something like...

```ruby
class Product < ActiveRecord::Base
  belongs_to :seller
  scope :total_stock_for, -> (seller) { where(seller: seller).sum(:stock) }
end
```